pkg/model/project.go

```diff
@@ -395,3 +395,57 @@ func ValidateUserGroups(groups []*ProjectUserGroup) error {
 	}
 	return nil
 }
+
+func (p *Project) MergeRBAC() *Project {
+	rbac := p.Rbac
+	if rbac == nil {
+		return p
+	}
+
+	size := len(p.UserGroups) + 3 // add the count of rbac teams.
+	v := make([]*ProjectUserGroup, 0, size)
+	if rbac.Admin != "" {
+		group := &ProjectUserGroup{
+			SsoGroup: rbac.Admin,
+			Role:     builtinRBACRoleAdmin.String(),
+		}
+		v = append(v, group)
+	}
+	if rbac.Editor != "" {
+		group := &ProjectUserGroup{
+			SsoGroup: rbac.Editor,
+			Role:     builtinRBACRoleEditor.String(),
+		}
+		v = append(v, group)
+	}
+	if rbac.Viewer != "" {
+		group := &ProjectUserGroup{
+			SsoGroup: rbac.Viewer,
+			Role:     builtinRBACRoleViewer.String(),
+		}
+		v = append(v, group)
+	}
+	v = append(v, p.UserGroups...)
+
+	p.UserGroups = v
+	return p
+}
+
+func (p *Project) WithBuiltinRBACRoles() *Project {
```

Comment:
I feel that we should avoid updating the Project model like this.
Instead, we can have a function like `ListAllRBACRoles() []ProjectRBACRole` which merges and returns the list of all custom and built-in roles.